internal/api/apiservice/smoketest_test.go

```diff
@@ -0,0 +1,166 @@
+package apiservice
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/oss-rebuild/internal/api"
+	"github.com/google/oss-rebuild/pkg/rebuild/rebuild"
+	"github.com/google/oss-rebuild/pkg/rebuild/schema"
+	"google.golang.org/grpc/codes"
+)
+
+func TestRebuildSmoketest(t *testing.T) {
+	tests := []struct {
+		name           string
+		request        schema.SmoketestRequest
+		smoketestStub  func(context.Context, schema.SmoketestRequest) (*schema.SmoketestResponse, error)
+		versionStub    func(context.Context, schema.VersionRequest) (*schema.VersionResponse, error)
+		expectedResp   *schema.SmoketestResponse
+		expectedErr    error
+		expectedErrMsg string
+	}{
+		{
+			name: "Successful smoketest",
+			request: schema.SmoketestRequest{
+				Ecosystem: "npm",
+				Package:   "test-package",
+				Versions:  []string{"1.0.0"},
+			},
+			smoketestStub: func(ctx context.Context, req schema.SmoketestRequest) (*schema.SmoketestResponse, error) {
+				return &schema.SmoketestResponse{
+					Executor: "1",
+					Verdicts: []schema.Verdict{
+						{
+							Target: rebuild.Target{
+								Ecosystem: rebuild.Ecosystem(req.Ecosystem),
+								Package:   req.Package,
+								Version:   req.Versions[0],
+							},
+							Message: "Success",
```

Comment:
nit: I think a success is an empty message, but it's important to make sure those messages make it through. Maybe change this to something other than "Success"?